File: Hashing/ts/mostFrequentChar.ts
export function mostFrequentChar(s: string): string {
  const counts: Record<string, number> = {};
  for (const char of s) {
    counts[char] = (counts[char] ?? 0) + 1;
  }

  let best = "";
  let bestCount = -1;
  for (const char of s) {
    const freq = counts[char];
    if (freq > bestCount) {
      best = char;
      bestCount = freq;
    }
  }

  return best;
}

File: Hashing/csharp/MostFrequentChar.cs
using System.Collections.Generic;

namespace HashingSolutions;

public static class MostFrequentChar
{
    public static char Solve(string input)
    {
        var counts = new Dictionary<char, int>();
        foreach (char c in input)
        {
            counts.TryGetValue(c, out int value);
            counts[c] = value + 1;
        }

        char best = '\0';
        var bestCount = -1;
        foreach (char c in input)
        {
            var current = counts[c];
            if (current > bestCount)
            {
                best = c;
                bestCount = current;
            }
        }

        return best;
    }
}
